src/ControlSystem/ControlErrors/SizeControlInfo.hpp

```diff
@@ -12,26 +14,75 @@
 ///   to SizeControlInfo::target_char_speed.
 /// - DeltaR: drives the minimum distance between the horizon and the excision
 ///   boundary to be constant in time.
-enum class SizeControlLabel { Initial, AhSpeed, DeltaR };
+/// - DeltaRDriftInward: Same as DeltaR but the excision boundary has a small
+///   velocity inward.  This state is triggered when it is deemed that the
+///   excision boundary and the horizon are too close to each other; the
+///   small velocity makes the excision boundary and the horizon drift apart.
+/// - DeltaRDriftOutward: Same as DeltaR but the excision boundary has a small
+///   velocity outward.  This state is triggered when it is deemed that the
+///   excision boundary and the horizon are too far apart.
+/// - DeltaRTransition: Same as DeltaR except for the logic that
+///   determines how DeltaRTransition changes to other states.
+///   DeltaRTransition is allowed (under some circumstances) to change
+///   to state DeltaR, but DeltaRDriftOutward and DeltaRDriftInward
+///   are never allowed to change to state DeltaR.  Instead
+///   DeltaRDriftOutward and DeltaRDriftInward are allowed (under
+///   some circumstances) to change to state DeltaRTransition.
+///
+/// The reason that DeltaRDriftInward, DeltaRDriftOutward, and
+/// DeltaRTransition are separate states is to simplify the logic.  In
+/// principle, all 3 of those states could be merged with state
+/// DeltaR, because the control error is the same for all four states
+/// (except for a velocity term that could be set to zero).  But if that
+/// were done, then there would need to be additional complicated
+/// logic in determining transitions between different states, and
+/// that logic would depend not only on the current state, but also on
+/// the previous state.
+enum class SizeControlLabel : size_t {
+  Initial,
+  AhSpeed,
+  DeltaR,
+  DeltaRDriftInward,
+  DeltaRDriftOutward,
+  DeltaRTransition
+};
 
 /// Holds information that is saved between calls of SizeControl.
 struct SizeControlInfo {
+  // SizeControlInfo needs to be serializable because it will be
+  // stored inside of a ControlError.
+  void pup(PUP::er& p) {
+    if(p.isUnpacking()) {
+      size_t state_as_size_t;
+      p | state_as_size_t;
+      state = static_cast<SizeControlLabel>(state_as_size_t);
+    } else {
+      auto state_as_size_t = static_cast<size_t>(state);
+      p | state_as_size_t;
+    }
```

Comment:
I think you can just `p | state;` for an enum.

Reply:
the enum goes away with my next commit.